### PR TITLE
All Service Node Keys as Base32z RPC Call

### PR DESCRIPTION
**Pull request by Doy-lee**
Base32z for interop with Lokinet

Example usage:
```
curl -X POST http://127.0.0.1:22023/json_rpc -d '{"jsonrpc":"2.0","id":"0","method":"get_all_service_nodes_keys", "params": {"fully_funded_nodes_only": false}' -H 'Content-Type: application/json'
```

Fully_funded_nodes_only is defaulted to true when not specified. Specifying it will ensure the returned keys are belonging to service nodes that have completed the staking requirement.

### CODE REVIEW

File: src/common/loki.cpp [hunk 1]
```diff
@@ -394,3 +394,157 @@ loki_exp2 (double x)
...
+    else
+      return nullptr;
+  }
+  return &stack[0];
+}
+
+std::string loki::hex64_to_base32z(const std::string &src)
+{
+  char buf[128]      = {};
+  std::string result = base32z_encode(src, buf);
```
Comment (msgmaxim) [inline src/common/loki.cpp]: Do we need to handle the case where `base32z_encode` returns a `nullptr`?

Comment (msgmaxim) [reply src/common/loki.cpp]: Also, kind of annoying that we throw away the buffer, can we use the buffer inside the string being returned instead?

File: src/cryptonote_core/cryptonote_core.h [hunk 2]
```diff
@@ -849,7 +849,7 @@ namespace cryptonote
       *
       * @param keys The container in which to return the keys
       */
-     void get_all_service_node_public_keys(std::vector<crypto::public_key>& keys) const;
+     void get_all_service_nodes_public_keys(std::vector<crypto::public_key>& keys) const;
```
Comment (msgmaxim) [inline src/cryptonote_core/cryptonote_core.h]: Reason for the change?

Comment (Doy-lee) [reply src/cryptonote_core/cryptonote_core.h]: Consistency with the other service node api

File: src/common/loki.cpp (hunk 1 above)
Comment (Doy-lee) [reply src/common/loki.cpp]: Added a nullptr check. string.data() returns const buffer and seem iffy about const casting that out.

**Pull request merged**